concent_api/core/models.py

```diff
@@ -73,3 +75,216 @@ class MessageAuth(Model):
+    delivered            = models.BooleanField(default = False)
+
+    subtask              = models.ForeignKey(Subtask)
+    created_at           = models.DateTimeField(default = get_current_utc_timestamp)
+
+
+class AbstractStoredMessage(models.Model):
+
+    timestamp = models.DateTimeField()
+    signature = Base64Field(max_length = 64)
+
+    class Meta:
+        abstract = True
+
+
+class StoredTaskToCompute(AbstractStoredMessage):
+
+    requestor_id                    = models.CharField(max_length = 32)
+    requestor_public_key            = Base64Field(max_length = 64)
+    requestor_ethereum_public_key   = Base64Field(max_length = 64)
+    provider_id                     = models.CharField(max_length = 32)
+    provider_public_key             = Base64Field(max_length = 64)
+    provider_ethereum_public_key    = Base64Field(max_length = 64)
+    compute_task_def                = JSONField()
+    package_hash                    = models.CharField(max_length = 32)
+    concent_enabled                 = models.BooleanField(default = False)
+
+
+class StoredReportComputedTask(AbstractStoredMessage):
+
+    subtask_id          = models.CharField(max_length = 32)
+    result_type         = models.PositiveSmallIntegerField()
+    computation_time    = models.IntegerField()
+    node_name           = models.CharField(max_length = 32)
+
+    #  Would it be IP address or what kind of address?
+    address             = models.CharField(max_length = 32)
+
+    #  Would it be a bytes params which indicate on information about resource owner?
+    node_info           = models.BinaryField()
+    port                = models.PositiveSmallIntegerField()
+    key_id              = models.CharField(max_length = 32)
+    extra_data          = JSONField()
+    eth_account         = Base64Field(max_length = 64)
+    task_to_compute     = models.ForeignKey(StoredTaskToCompute)
+    size                = models.IntegerField()
+    checksum            = models.CharField(max_length = 32)
+
+
+class StoredTaskFailure(AbstractStoredMessage):
+
+    subtask_id      = models.CharField(max_length = 32)
+
+    #  What form would have this param?
+    err             = models.CharField(max_length = 32)
+    task_to_compute = models.ForeignKey(StoredTaskToCompute)
+
+
+class StoredCannotComputeTask(AbstractStoredMessage):
+
+    class CannotComputeReason(ChoiceEnum):
+        WrongCTD            = 'wrong_ctd'
+        WrongKey            = 'wrong_key'
+        WrongAddress        = 'wrong_address'
+        WrongEnvironment    = 'wrong_environment'
+        NoSourceCode        = 'no_source_code'
+        WrongDockerImages   = 'wrong_docker_images'
+
+    subtask_id      = models.CharField(max_length = 32)
+    task_to_compute = models.ForeignKey(StoredTaskToCompute)
+    reason          = models.CharField(max_length = 32, choices = CannotComputeReason.choices())
+
+
+class StoredAckReportComputedTask(AbstractStoredMessage):
```

Comment:
For each of these tables you should have an `assert` that checks that the fields in the model are the same as the fields in the Golem message. Otherwise we we'll have to manually check on every release if anything has been added or removed.